Documentation/contributing/development/contributing_guide.rst

```diff
@@ -47,8 +47,14 @@ upstream GitHub repository at https://github.com/cilium/cilium.
 Before hitting the submit button, please make sure that the following
 requirements have been met:
 
-#. Each commit compiles and is functional on its own to allow for bisecting of
-   commits.
+#. Take some time to describe your change in the PR description! A well-written
```

Comment:
I wonder whether we should explicitly suggest the same for the commit message?

I've seen contributors describing their change only in the PR description but keep very sparse or almost empty commit messages. This is especially true for PRs involving multiple commits. IMO keeping the motivation for a particular change in the git commit log as well helps tracking history of a change without having to rely on external information from GitHub. Also, mapping commits to the PR that introduced them involves additional steps on the GitHub UI.

Reply:
A little lower there is a dedicated step for commit messages. Do you think we should change/amend that step?

```
#. All commits contain a well written commit description including a title,                                                                                                                                                                                                                                                    
   description and a ``Fixes: #XXX`` line if the commit addresses a particular                                                                                                                                                                                                                                                 
   GitHub issue. Note that the GitHub issue will be automatically closed when                                                                                                                                                                                                                                                  
   the commit is merged.                                                                                                                                                                                                                                                                                                       
                                                                                                                                                                                                                                                                                                                               
   ::                                                                                                                                                                                                                                                                                                                          
                                                                                                                                                                                                                                                                                                                               
        apipanic: Log stack at debug level                                                                                                                                                                                                                                                                                     
                                                                                                                                                                                                                                                                                                                               
        Previously, it was difficult to debug issues when the API panicked                                                                                                                                                                                                                                                     
        because only a single line like the following was printed:                                                                                                                                                                                                                                                             
                                                                                                                                                                                                                                                                                                                               
        level=warning msg="Cilium API handler panicked" client=@ method=GET                                                                                                                                                                                                                                                    
        panic_message="write unix /var/run/cilium/cilium.sock->@: write: broken                                                                                                                                                                                                                                                
        pipe"                                                                                                                                                                                                                                                                                                                  
                                                                                                                                                                                                                                                                                                                               
        This patch logs the stack at this point at debug level so that it can at                                                                                                                                                                                                                                               
        least be determined in developer environments.                                                                                                                                                                                                                                                                         
                                                                                                                                                                                                                                                                                                                               
        Fixes: #4191                                                                                                                                                                                                                                                                                                           
                                                                                                                                                                                                                                                                                                                               
        Signed-off-by: Joe Stringer <joe@cilium.io>                                                                                                                                                                                                                                                                            
                                                                                                                                                                                                                                                                                                                               
   .. note::                                                                                                                                                                                                                                                                                                                   
                                                                                                                                                                                                                                                                                                                               
       Make sure to include a blank line in between commit title and commit                                                                                                                                                                                                                                                    
       description.
```

Reply:
Oh sorry, I must have overlooked that section in the diff view when reviewing the patch. I think the instructions for the commit message are perfectly clear as is and don't need change.